### PR TITLE
precomit file for git guardian usage

### PR DESCRIPTION
**Pull request by Baalmart**
# precommit file for git guardian usage

**_WHAT DOES THIS PR DO?_**
pre-commit file for git guardian usage.
Effort to carry out security checks on personal machines before pushing to Github.

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

N/A

**_ARE THERE ANY RELATED PRs?_**

N/A

### CODE REVIEW

**Review by 123MwanjeMike — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+repos:
```
Comment (Baalmart) [inline .pre-commit-config.yaml:1]: In relation to Git guardian and pre-commit checks, I am curious to find out why my commits always fail. I always have to add the -n flag to be able to commit. Attached is a screenshot which demonstrates these commit failures. @danielogen 

- https://docs.gitguardian.com/internal-repositories-monitoring/integrations/git_hooks/pre_commit?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment

Screenshot 2022-03-01 at 16 23 17

Comment (danielogen) [reply .pre-commit-config.yaml:1]: @Baalmart , I am very sure. The `-n` flag is used to skip the precommit check. Let me test it out see.

**Review by danielogen — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Baalmart) [reply .pre-commit-config.yaml:1]: Thanks @danielogen , just wondering the relevance if I am going to always skip this check. 🤔 cc: @123MwanjeMike

Comment (danielogen) [reply .pre-commit-config.yaml:1]: True @Baalmart , that wouldn't make sense. I guess there is something we are missing!

Comment (123MwanjeMike) [reply .pre-commit-config.yaml:1]: I think we need to do more investigation into it

**Review by NoahNsimbe — APPROVED** [review]
LGTM!..looing forward to its usage

File: .pre-commit-config.yaml (hunk 1 above)
Comment (123MwanjeMike) [reply .pre-commit-config.yaml:1]: To fix this, you need need to [create an API key](https://docs.gitguardian.com/internal-repositories-monitoring/gg_shield/getting_started#step-1-authenticate-using-an-api-key) and then add it to the **_.env_** file at the root of the directory (same location as .pre-commit-config.yaml) or in your **_~/.bash_profile_** file

_/.env_
`GITGUARDIAN_API_KEY=yourGeneratedAPIkeyHere`

 cc @Baalmart @danielogen

Comment (Baalmart) [reply .pre-commit-config.yaml:1]: let me try it out asap. Thanks @123MwanjeMike

**Pull request merged**